src/client.rs

```diff
@@ -250,6 +253,36 @@ impl Client {
     {
         Api::namespaced(self.client.clone(), namespace)
     }
+
+    /// Waits until given resource with metadata is recognized as applied by Kubernetes API
+    ///
+    /// # Arguments
+    ///
+    /// - `namespace` - Optional namespace to look for the resources in.
+    /// - `lp` - Parameters to filter resources in given namespace.
+    pub async fn wait_ready<T>(&self, namespace: Option<String>, lp: ListParams)
+    where
+        T: Meta + Clone + DeserializeOwned + Send + 'static,
+    {
+        let api: Api<T> = self.get_api(namespace);
+        let mut watcher = kube_runtime::watcher(api, lp).boxed();
+
+        while let Some(result) = watcher.next().await {
+            match result {
+                Ok(event) => match event {
+                    Event::Applied(_) | Event::Restarted(_) => {
+                        break;
+                    }
+                    Event::Deleted(_) => {
```

Comment:
Can this end in an endless wait/loop?
If I create a resource, immediately delete it and then call `wait_ready` (for whatever reason) could it be that I only receive the `Deleted` Event?
Thinking out loud: I guess not because the first thing watcher does is to list all current resources which'd mean we'd see it before it's deleted.

Reply:
Yes it can, unles there's a timeout set up by the caller on this whole `wait_ready` function. Not sure if this is a good pattern, but using e.g. tokio, you can always timeout futures yourself, no need for this function to do it for you. I use it 
on this line: https://github.com/stackabletech/operator-rs/pull/79/files#diff-31b064fae044a3049a84f1d3379839a417abdbf581ef11e516aa2a94eeb34a40R97